siliconcompiler/remote/server.py

```diff
@@ -323,10 +324,25 @@ async def remote_sc(self, chip, username):
             # Run the job with slurm clustering.
             chip.set('option', 'scheduler', 'name', 'slurm')
 
-        chip.run()
+        # Run one task at a time, archiving each node after completion.
+        steplist = chip.get('option', 'steplist')
+        for step in steplist:
+            chip.set('option', 'steplist', step)
+            chip.run()
+            indexlist = chip.getkeys('flowgraph', chip.get('option', 'flow'), step)
+            for index in indexlist:
+                chip.cwd = os.path.join(chip.get('option', 'builddir'), '..')
+                tf = tarfile.open(os.path.join(self.nfs_mount,
+                                               job_hash,
+                                               f'{job_hash}_{step}{index}.tar.gz'),
+                                  mode='w:gz')
+                chip._archive_node(tf, step=step, index=index)
+                tf.close()
+        chip.set('option', 'steplist', steplist)
 
         # Create a single-file archive to return if results are requested.
-        with tarfile.open(os.path.join(self.nfs_mount, f'{job_hash}.tar.gz'), "w:gz") as tar:
+        # TODO: Remove after API updates go out.
+        with tarfile.open(os.path.join(self.nfs_mount, job_hash, f'{job_hash}_.tar.gz'), "w:gz") as tar:
```

Comment:
why is this behavior different? it seems like we would want server.py to have the proper API implemented.

Reply:
Storing a whole-job archive makes the server backwards-compatible with older clients - once the API update is out on pypi, we can safely remove this.

Although, since the code is in the SC repo's development server, we can probably just remove it now.